veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,314 @@
+A few artifacts are required for the deployment of the network binding plugin:
+- A container image build and publish process for the sidecar container.
+- A static daemonset manifest which serves to deploy the CNI.
+- A static manifest for applying a `network-attachment-definition` CR named `netbindingpasst` referencing to the CNI.
+[CNAO](https://github.com/kubevirt/cluster-network-addons-operator) will be enhanced to store and automate the deployment process.
+
+## API Examples
+
+No API changes are expected in kubevirt/kubevirt.
+### passt network binding plugin configuration in KubeVirt CR
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    network:
+      binding:
+        passt:
+          computeResourceOverhead:
+            requests:
+              memory: 250MiB
+          migration: {}
+          networkAttachmentDefinition: default/netbindingpass
+          sidecarImage: 
+            registry: quay.io/kubevirt/network-passt-binding:<tag/sha>
+```
+
+### VM network interface with passt network binding 
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+spec:
+  template:
+    spec:
+      domain:
+        devices:
+          interfaces:
+          - name: passtnet
+            binding:
+              name: passt
+            ports:
+            - name: http
+              port: 80
+              protocol: TCP
+```
+
+## Alternatives
+
+TCP live migration requires a privileged action (TCP_REPAIR socket option). A straightforward approach is granting passt the required capability (`CAP_NET_ADMIN`). Ambient capabilities can be applied to the passt binary similarly to [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/cmd/virt-launcher-monitor/virt-launcher-monitor.go#L176).
+
+However, container runtimes drop all capabilities from the [bounding set](https://man7.org/linux/man-pages/man7/capabilities.7.html), and capabilities must explicitly be added in the SecurityContext, as done for CAP_NET_BIND_SERVICE in [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-controller/services/rendercontainer.go#L284).
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
```

Comment:
Is there a way (other than rebooting the VM) to upgrade an existing VM using the "old" version of  the passt BP to the newer version? As there will be differences in the domain XML.

Please consider adding a bullet about the incompatibility of the new passt BP with older versions of KubeVirt.

What will happen to existing users of the passt BP wanting to upgrade their BP version but do not have a networking provider that supports sticky IP?